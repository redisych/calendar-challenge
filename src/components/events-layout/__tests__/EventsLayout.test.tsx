import * as React from 'react';
import * as enzyme from 'enzyme';
import { EventsLayout } from '..';
import toJson from 'enzyme-to-json';

const getWrapper = (): any => {
    return enzyme.shallow(
        <EventsLayout />
    );
};

describe('EventsLayout', () => {
    it('should be defined', () => {
        const wrapper = getWrapper();

        expect(wrapper).toBeDefined();
    });

    it('should match snapshot', () => {
        const wrapper = getWrapper();

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should listen to layOutDay event', () => {
        const fakeLayOutListener = jest.fn((event, callback) => {
            const fakeEvent = new CustomEvent(
                event,
                {
                    detail: {
                        calendarEvents: []
                    }
                }
            );
            callback(fakeEvent);
        });

        fakeLayOutListener.mock.calls = [];
        document.addEventListener = fakeLayOutListener;

        getWrapper(); 
        expect(fakeLayOutListener.mock.calls.length).toBe(1);
    });

    it('should correctly calculate children clusters count', () => {
        const mockCalendarEvents = [{start: 0, end: 60}, {start: 60, end: 100}];
        const fakeLayOutListener = jest.fn((event, callback) => {
            const fakeEvent = new CustomEvent(
                event,
                {
                    detail: {
                        calendarEvents: mockCalendarEvents
                    }
                }
            );
            callback(fakeEvent);
        });

        document.addEventListener = fakeLayOutListener;

        const wrapper = getWrapper();
        const eventsClusters = wrapper.state().eventsClusters;
        expect(eventsClusters).toHaveLength(2);
        expect(wrapper.children()).toHaveLength(2);
    });
});
