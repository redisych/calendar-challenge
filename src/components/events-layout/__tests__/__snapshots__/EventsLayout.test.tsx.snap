// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsLayout should match snapshot 1`] = `
<div
  className="events-layout"
  style={
    Object {
      "height": 720,
      "minWidth": 620,
      "width": 620,
    }
  }
/>
`;
